testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,526 @@
+            try {
+                Field[] declaredFields = c.getDeclaredFields();
+                Field[] fields = c.getFields();
+                allFields.addAll(Arrays.asList(declaredFields));
+                allFields.addAll(Arrays.asList(fields));
+            } catch (NoClassDefFoundError e) {
+                e.printStackTrace();
+                continue;
+            } catch (Exception exception) {
+                exception.printStackTrace();
+                continue;
+            }
+            // prepare for the subxml fold
+
+            for (Field f : allFields) {
+                String fieldName = getFieldFQN(f);
+
+                // if a field is final and has a primitive type there's no point to capture it.
+                if (Modifier.isStatic(f.getModifiers())
+                    && !(Modifier.isFinal(f.getModifiers()) &&  f.getType().isPrimitive())) {
+                    try {
+                        if (shouldCapture(f)) {
+                            allFieldName.add(fieldName);
+                            f.setAccessible(true);
+
+                            Object instance;
+                            try {
+                                instance = f.get(null);
+                            } catch (NoClassDefFoundError ncdfe) {
+                                instance = null;
+                                ncdfe.printStackTrace();
+                            }
+                            if (instance instanceof ThreadLocal) {
+                                instance = ((ThreadLocal)instance).get();
+                            }
+                            LinkedHashMap<String, Object> nameToInstance_temp = new LinkedHashMap<String, Object>();
+                            nameToInstance_temp.put(fieldName, instance);
+
+                            dirty = false;
+                            serializeRoots(nameToInstance_temp);
+                            if (!dirty) {
+                                nameToInstance.put(fieldName, instance);
+
+                                String ob4field = serializeOBs(instance);
+                                writer = new PrintWriter(subxmlDir + "/" + fieldName + ".xml", "UTF-8");
+                                writer.println(ob4field);
+                                writer.close();
+                            }
+                        }
+                    } catch (OutOfMemoryError ofme) {
+                        ofme.printStackTrace();
+                        continue;
+                    } catch (NoClassDefFoundError ncdfe) {
+                        ncdfe.printStackTrace();
+                        continue;
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                        continue;
+                    }
+                }
+            }
+        }
+
+        writer = new PrintWriter(Configuration.config().getProperty("statecapture.allFieldsFile"), "UTF-8");
```

Comment:
Could it be possible that user forgets to provide this file name? Would that crash things? Should probably check for that.